homeassistant/components/vizio/media_player.py

```diff
@@ -83,18 +106,30 @@
             "is valid and available, device type is correct%s",
             fail_auth_msg,
         )
-        return
+        raise PlatformNotReady
+
+    entity = VizioDevice(
+        config_entry, device, name, volume_step, device_type, unique_id
+    )
 
-    async_add_entities([VizioDevice(device, name, volume_step, device_type)], True)
+    async_add_entities([entity], True)
 
 
 class VizioDevice(MediaPlayerDevice):
     """Media Player implementation which performs REST requests to device."""
 
     def __init__(
-        self, device: VizioAsync, name: str, volume_step: int, device_type: str
+        self,
+        config_entry: ConfigEntry,
+        device: VizioAsync,
+        name: str,
+        volume_step: int,
+        device_type: str,
+        unique_id: str,
```

Comment:
Since we have the unique_id on the entry, we don't need to pass the unique_id separately.